src/xenia/memory.cc

```diff
@@ -516,6 +546,82 @@ void BaseHeap::DumpMap() {
   }
 }
 
+bool BaseHeap::Save(ByteStream* stream) {
+  XELOGD("Heap %.8X-%.8X", heap_base_, heap_base_ + heap_size_);
+
+  for (size_t i = 0; i < page_table_.size(); i++) {
+    auto& page = page_table_[i];
+    stream->Write(page.qword);
+    if (!page.state) {
+      // Unallocated.
+      continue;
+    }
+
+    // TODO: Write compressed
+    if (page.state & kMemoryAllocationCommit &&
+        page.current_protect & kMemoryProtectRead) {
```

Comment:
we may have to write pages even with noaccess - as a game could put valid data somewhere, Protect() it with no access, and then unprotect it later and expect it to be there.